src/NuGet.Clients/NuGet.Credentials/ICredentialProvider.cs

```diff
@@ -5,18 +5,20 @@
 using System.Net;
 using System.Threading;
 using System.Threading.Tasks;
+using NuGet.Configuration;
 
 namespace NuGet.Credentials
 {
     public interface ICredentialProvider
     {
         string Id { get; }
 
-        Task<CredentialResponse> Get(Uri uri, 
-                               IWebProxy proxy, 
-                               bool isProxyRequest, 
-                               bool isRetry,
-                               bool nonInteractive, 
-                               CancellationToken cancellationToken);
+        Task<CredentialResponse> Get(
+            Uri uri, 
+            IWebProxy proxy, 
+            CredentialRequestType type, 
```

Comment:
probably a good idea to pass a default message, so providers don't have to repeat the code in the console provider